### DISCUSSION TITLE
RTK: Wi-Fi connection to Reach M2 takes very long + RTK regularly not cm precise

### DISCUSSION THREAD

**Discussion by bconvens** [ctu-mrs/mrs_uav_system#89] [Q&A]
Hi,

I have been testing the reach M2 module and it takes - not always but very often - very long (>5 minutes) to make a connection to my router.
The reach M2 firmware was updated. Sometimes unplugging and plugging the reach M2 USB helps, but not always. But this is a bit annoying since then I have to run back to the drone multiple times. 
On the other hand, the Reach RS2 always connects directly without any issues so far. 
If the reach M2 finally gets connected, the RTK system seems to work.
Although I have to say, when hovering with a F450 drone (MpcTracker, Se3Controller) there where some tests (same test location, same moment of the day, same open sky, ...) where the drone was hovering precisely on a single spot +- 0.1m error (which is good enough for me) and some test +- 1.5m (making some kind of circles or drifting away in a certain direction).

Did you have the same issue(s) in the past and potential solutions (e.g. config settings) I could try? 
Although I think they are unrelated, do you think these two issues might be related somehow (e.g. a malfunctioning reachm2 module?)
Is there a specific order of device booting to respect during the experiments to make the reach M2 connect immediately?

Thanks!

**Comment by DanHert** [comment]
Hello, regarding the RTK localization, there was an issue in our RTK estimator, but this issue has been fixed, so if you pull uav_core and try flying, the localization and control should be very precise (Of course only when you have a RTK fix).

As for the M2, we cannot really help you with that, we use the LoRa for corrections and Serial->USB to get the data to the NUC, we do not connect to the wifi for operation. 

Dan H.
